python-lib/dku_email_client.py

```diff
@@ -0,0 +1,112 @@
+import dataiku
+import logging
+from abc import ABC, abstractmethod
+from collections import namedtuple
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+
+
+def SmtpConfig():
+    # str, int, bool, bool, str, str
+    return namedtuple("SmtpConfig", "smtp_host, smtp_port, smtp_use_tls, smtp_use_auth, smtp_user, smtp_pass")
+
+
+def AttachmentFile():
+    """ Format for an attachment file with the info broken down nicely for both the client classes """
+    # str, str, str ('application' or 'text'), bytes
+    return namedtuple("AttachmentFile", "file_name, mime_type, mime_subtype, data")
+
+
+class AbstractMessageClient(ABC):
+    def __init__(self, plain_text):
+        self.plain_text = plain_text
+
+
+    def send_email(self, sender, recipient, email_subject, email_body, attachment_files):
+        """
+        :param sender: sender email, str - depending on impl, could be ignored if a sender configured for the channel
+        :param recipient: recipient email, str
+        :param email_subject: str
+        :param email_body: body of either plain text or html, str
+        :param attachment_files:attachments as list of  AttachmentFile
+        """
+        # Any generic email body decoration goes here
+        email_body_to_send = email_body
+        if self.plain_text:
+            email_body_to_send = email_body + '\n\n'
```

Comment:
This should be done in the `build_email`method or in the `send_mail_for_conrtact` method. 
Message client should only take care of sending already formatted emails, not format or alter the body.
Plus if you remove it, you don"t need a `send_email_impl` method, only an abstract `send_email` method

Reply:
Done [here](https://github.com/dataiku/dss-plugin-sendmail/pull/8/commits/546ecdc32cd570ce50ffa5cb7109db18bb2e1858)